external-crates/move/crates/move-compiler/src/command_line/compiler.rs

```diff
@@ -96,21 +114,44 @@ enum PassResult {
     Typing(typing::ast::Program),
     HLIR(hlir::ast::Program),
     CFGIR(cfgir::ast::Program),
-    Compilation(Vec<AnnotatedCompiledUnit>, /* warnings */ Diagnostics),
+    Compilation(
+        Vec<AnnotatedCompiledUnit>,
+        Vec<ModuleIdent>,
+        /* warnings */ Diagnostics,
+    ),
 }
 
+pub type CompiledModuleInfoMap = BTreeMap<ModuleIdent, Arc<CompiledModuleInfo>>;
+
 #[derive(Clone)]
-pub struct FullyCompiledProgram {
-    pub files: MappedFiles,
-    pub parser: parser::ast::Program,
-    pub expansion: expansion::ast::Program,
-    pub naming: naming::ast::Program,
-    pub typing: typing::ast::Program,
-    pub hlir: hlir::ast::Program,
-    pub cfgir: cfgir::ast::Program,
-    pub compiled: Vec<AnnotatedCompiledUnit>,
+pub struct CompiledModuleInfo {
+    pub file_name: Symbol,
+    pub file_content: Arc<str>,
+    /// to compute address conflicts in expansion/translate.rs
+    /// (address map "inlined" into the module so that
+    /// modules are separated)
+    pub named_address_map: NamedAddressMap,
```

Comment:
These are usually pretty small, but I wonder if we should Arc/Rc them

Reply:
That's totally fair!